roundrobin/stickysessions.go

```diff
@@ -81,15 +80,29 @@ func (s *StickySession) StickBackend(backend *url.URL, w *http.ResponseWriter) {
 	http.SetCookie(*w, cookie)
 }
 
-func (s *StickySession) isBackendAlive(needle *url.URL, haystack []*url.URL) bool {
+func (s *StickySession) isBackendAlive(needle string, haystack []*url.URL) (bool, *url.URL) {
 	if len(haystack) == 0 {
-		return false
+		return false, nil
 	}
 
-	for _, serverURL := range haystack {
-		if sameURL(needle, serverURL) {
-			return true
+	switch {
+	case strings.HasPrefix(needle, "http"):
+		for _, serverURL := range haystack {
+			if needle == serverURL.String() {
```

Comment:
`sameURL` use to check only for `Path`, `Host` and `Scheme` equality. If this change is intended could you tell us more?

Reply:
It's either creating a net.URL out of the `needle` string to compare it with `serverURL` with `sameURL(...)` or Marshaling serverURL into a string to compare it with needle.

I guess both are pretty similar.

Reply:
It's really except that with `.String()` you will also compare the `User` if present.

Reply:
Ok, I did not know it was a thing to discard the user info but ok.